gobblin-utility/src/main/java/org/apache/gobblin/util/ConfigUtils.java

```diff
@@ -67,12 +70,48 @@
    */
   private static final String GOBBLIN_CONFIG_BLACKLIST_KEYS = "gobblin.config.blacklistKeys";
 
+  /**
+   * Gobblin sync system specific config
+   */
+  public static final String GOBBLIN_SYNC_SYSTEMS_KEY = "gobblin_sync_systems";
+  public static final String GOBBLIN_SYNC_SYSTEM_CONFIG_FILES_KEY = "config_files";
+  public static final String GOBBLIN_SYNC_SYSTEM_METASTORE_URI_KEY = "metastore.uris";
+  public static final String IS_SECURE_KEY = "is_secure";
+  public static final String IS_TOKEN_MGMT_ENABLED_KEY = "is_token_management_enabled";
+  public static final String CONFIG_FILES_LIST_KEY = "config_files";
+  public static final String HIVE_SYSTEM_KEY = "hive";
+  public static final String HIVE_METASTORE_URI_KEY = "metastore.uris";
+
+
   /**
    * A suffix that is automatically appended to property keys that are prefixes of other
    * property keys. This is used during Properties -> Config -> Properties conversion since
    * typesafe config does not allow such properties. */
   public static final String STRIP_SUFFIX = ".ROOT_VALUE";
 
+  private static volatile Config instance;
+  public static Config getInstance() {
```

Comment:
the empty wont have the config from application.conf and i notice we keep loading config many times in code. so this method should set the base for all initial config and provide the singleton instead of loading the config multiple times.